extras/fuzz_headers.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+
+import sys
+
+import atheris
+
+import scrapy
+from scrapy.crawler import CrawlerProcess
+
+
+def TestOneInput(data):
+    fdp = atheris.FuzzedDataProvider(data)
+    test = fdp.ConsumeUnicodeNoSurrogates(fdp.ConsumeIntInRange(0, 4096))
+
+    class test_spider(scrapy.Spider):
+        name = "test_spider"
+
+        url = "http://quotes.toscrape.com/page/1/"
+
+        def start_requests(self):
+            headers = {"User-Agent": test, "Accept-Language": test, "Referer": test}
+
+            yield scrapy.Request(self.url, headers=headers, callback=self.parse)
+
+        def parse(self, response):
+            pass
+
+    try:
+        process = CrawlerProcess()
+        process.crawl(test_spider)
+        process.start()
+    except Exception as e:
+        print(e)
```

Comment:
Don’t you get a [reactor not restartable issue](https://stackoverflow.com/questions/41495052/scrapy-reactor-not-restartable) when running this code multiple times on the same process?

Reply:
They told me it indeed does.